include/yaramod/types/expressions.h

```diff
@@ -400,23 +409,36 @@ class UnaryOpExpression : public Expression
 
 protected:
 	template <typename ExpPtr>
-	UnaryOpExpression(TokenIt op, ExpPtr&& expr)
+	UnaryOpExpression(TokenIt op, ExpPtr&& expr, bool operatorLeft)
```

Comment:
Over the years I kinda refrained from boolean parameters/arguments to more strictly typed values as it provides more explainability and type security. So instead of having `bool operatorLeft` I'd welcome something in lines of

```c++
enum class UnaryOperatorPlacement
{
    Left,
    Right
}
```

Reply:
Also if you want to push this idea even further, you can have

```c++
template <typename ExpPtr, UnaryOperatorPlacement OpPlacement>
class UnaryOpExpression { ... }
```

and then you can directly inherit from `UnaryOpExpression<_, UnaryOpPlacement::Left>` or `UnaryOpExpression<_, UnaryOpPlacement::Right>`. You'll avoid having if's in your `getText` and you can still have very simple constructor.

Reply:
Thank you very much for the tip. I tried to push it further, so I introduced
```
template <typename ExpPtr, UnaryOperatorPlacement OpPlacement>
class UnaryOpExpression : public Expression
```
but I don't know how to create two virtual methods `getText` for both options and I also struggle with the inheritance:
```
error: ‘_’ was not declared in this scope:
class DefinedExpression : public UnaryOpExpression<_, UnaryOperatorPlacement::Left>
```
I will try more tomorrow.

Reply:
Right, in that case, don't use the second approach. I forgot that it's all implemented in virtual method.

Reply:
Done, thank you.

Reply:
Done, thank you!